.github/workflows/mergify-ready.yml

```diff
@@ -88,3 +88,32 @@ jobs:
             echo -e "$fixup_commits"
             exit 1
           fi
+
+  no-fixup-commits:
+    runs-on: ubuntu-latest
+    if: >-
+      github.event_name == 'pull_request' &&
+      github.event.pull_request.draft == false &&
+      github.event.pull_request.base.ref == 'master' && 
+      contains(github.event.pull_request.labels.*.name, 'automerge:rebase') &&
+      !contains(github.event.pull_request.labels.*.name, 'bypass:linear-history')
+
+    env:
+      HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+      BASE_SHA: ${{ github.event.pull_request.base.sha }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check for fixup commits
+        id: fixup-commits
+        run: |
+          if [[ $(git rev-list "$BASE_SHA".."$HEAD_SHA" --grep="^fixup! " | wc -l) -eq 0 ]]; then
```

Comment:
I just realized we're only checking `fixup!` commits here, and we should also check for `squash!` and `amend!` commits as well. As a drive-by, we should fix the `linear-history` job above to include `amend!` too